Make empty menu an error

Rembane opened the pull request:
Before this change the empty menu could either be represented by `Left NoLunch` or `Right []`, which introduces an unnecessary ambiguity. This commit does not let a menu be empty by using the excellent `NonEmpty` type, and then doing all the necessary changes in the code base.

Rembane committed:
Make empty menu an error

Before this change the empty menu could either be represented by `Left
NoLunch` or `Right []`, which introduces an unnecessary ambiguity. This
commit does not let a menu be empty by using the excellent `NonEmpty`
type, and then doing all the necessary changes in the code base.

Jassob reviewed (commented):
Looks solid enough to me!

The1Penguin commented on src/Util.hs:
```diff
@@ -22,11 +24,10 @@ import           Lens.Micro.Platform
 menusToEitherNoLunch = \case
   [] -> Left NoLunch
-  xs -> Right xs
+  xs -> (Right . NE.fromList) xs
```
This is fine, but I am thinking if one should be explicitly using the constructor for `NonEmpty`... It doesn't really introduce any potential issues doing it this way however

Rembane replied:
Woop woop! Then we need to deconstruct `xs` and then construct the new thing like this: `(x:xs) -> Right (x :| xs)` is this your intention? :smiley_cat:

The1Penguin replied:
Yes :smile_cat:

Rembane replied:
Sweet! I think the code got prettier with your suggestions: https://github.com/dtekcth/mat-chalmers/pull/154/commits/f0ec07b0b8d9db7b05585aea7b6a4be4bbb2894d I'll squash this before merge.

The1Penguin reviewed (approved):
Think about what I mentioned in my other comment, it doesn't really matter either way. Good idea :D

Rembane committed:
Use NonEmpty list constructor

...after feedback from Pingu. This lets us remove an import if we're
ready to deconstruct the list and then reconstruct it as a non empty
list. It produces surprisingly pretty code.

Rodeby1 reviewed (approved):
lgtm
